code/modules/research/rdconsole.dm

```diff
@@ -590,64 +625,67 @@ doesn't have toxins access.
-		else
-			l += "<h3><span class='linkOff bad'>Not Enough Points</span></h3>[RDSCREEN_NOBREAK]"
-	else if(stored_research.visible_nodes[selected_node.id])
-		l += "<h3><span class='linkOff bad'>Prerequisites not met!</span></h3>[RDSCREEN_NOBREAK]"
-	else
-		l += "<h3><span class='linkOff bad'>ERROR</span></h3>[RDSCREEN_NOBREAK]"
-	l += "<div><h3>Designs:</h3>[RDSCREEN_NOBREAK]"
-	for(var/i in selected_node.designs)
-		var/datum/design/D = selected_node.designs[i]
-		l += "<A href='?src=[REF(src)];view_design=[i]'>[D.name]</A>"
-	l += "</div><div><h3>Prerequisites:</h3>[RDSCREEN_NOBREAK]"
-	for(var/i in selected_node.prerequisites)
-		var/datum/techweb_node/prereq = selected_node.prerequisites[i]
-		var/sc = stored_research.researched_nodes[prereq.id]
-		var/begin
-		var/end
-		if(sc)
-			begin = "<font color='green'><b>"
-			end = "</font></b>"
-		else
-			begin = "<span class='bad'>"
-			end = "</span>"
-		l += "<A href='?src=[REF(src)];view_node=[i]'>[begin][prereq.display_name][end]</A>"
-	l += "</div><div><h3>Unlocks:</h3>[RDSCREEN_NOBREAK]"
-	for(var/i in selected_node.unlocks)
-		var/datum/techweb_node/unlock = selected_node.unlocks[i]
-		l += "<A href='?src=[REF(src)];view_node=[i]'>[unlock.display_name]</A>"
+		return
 
-	l += "</div>[RDSCREEN_NOBREAK]"
+	l += "<table><tr>[RDSCREEN_NOBREAK]"
+	if (length(selected_node.prerequisites))
+		l += "<th align='left'>Requires</th>[RDSCREEN_NOBREAK]"
+	l += "<th align='left'>Current Node</th>[RDSCREEN_NOBREAK]"
+	if (length(selected_node.unlocks))
+		l += "<th align='left'>Unlocks</th>[RDSCREEN_NOBREAK]"
+
+	l += "</tr><tr>[RDSCREEN_NOBREAK]"
+	if (length(selected_node.prerequisites))
+		l += "<td valign='top'>[RDSCREEN_NOBREAK]"
+		for (var/i in selected_node.prerequisites)
+			l += ui_techweb_single_node(selected_node.prerequisites[i])
+		l += "</td>[RDSCREEN_NOBREAK]"
+	l += "<td valign='top'>[RDSCREEN_NOBREAK]"
+	l += ui_techweb_single_node(selected_node, selflink=FALSE)
+	l += "</td>[RDSCREEN_NOBREAK]"
+	if (length(selected_node.unlocks))
+		l += "<td valign='top'>[RDSCREEN_NOBREAK]"
+		for (var/i in selected_node.unlocks)
+			l += ui_techweb_single_node(selected_node.unlocks[i])
+		l += "</td>[RDSCREEN_NOBREAK]"
+
+	l += "</tr></table>[RDSCREEN_NOBREAK]"
 	return l
 
 /obj/machinery/computer/rdconsole/proc/ui_techweb_designview()		//Legacy code
 	RDSCREEN_UI_SDESIGN_CHECK
 	var/list/l = list()
 	var/datum/design/D = selected_design
-	l += "<div>Name: [D.name]"
+	l += "<div>[build_path_icon(D.build_path)] <b>[D.name]</b>"
 	if(D.build_type)
-		l += "Lathe Types:"
-		if(D.build_type & IMPRINTER) l += "Circuit Imprinter"
-		if(D.build_type & PROTOLATHE) l += "Protolathe"
-		if(D.build_type & AUTOLATHE) l += "Autolathe"
-		if(D.build_type & MECHFAB) l += "Exosuit Fabricator"
-		if(D.build_type & BIOGENERATOR) l += "Biogenerator"
-		if(D.build_type & LIMBGROWER) l += "Limbgrower"
-		if(D.build_type & SMELTER) l += "Smelter"
-	l += "Required Materials:"
+		var/lathes = ""

```

Comment:
list.

Reply:
Performance concern? They're supposed to go on the same line.

Reply:
They would still go on the same line in a list
in DM listA += listB just puts list B's contents into listA

Reply:
Right, forgot about the big picture. Addressed.